docs/components/best-practices/security/csrf-protection.md

```diff
@@ -0,0 +1,115 @@
+
+For a comprehensive understanding of CSRF attacks and prevention methods, refer to the
+[MDN Web Docs on CSRF](https://developer.mozilla.org/en-US/docs/Glossary/CSRF).
+
+## How CSRF protection works in Camunda
+
+- **Token generation**: A unique CSRF token is generated and stored in a secure, HTTP-only cookie named `X-CSRF-TOKEN`.
+- **Token validation**: For state-changing requests (POST, PUT, DELETE, etc.), the server validates that the CSRF token
+  in the request header matches the one in the cookie.
+- **Safe methods**: GET, HEAD, TRACE, and OPTIONS requests are considered safe and don't require CSRF validation.
+
+## Configuration
+
+### Enabling/disabling CSRF protection
+
+CSRF protection is enabled by default for security. You can configure it using:
+
+**Application property:**  
+`camunda.security.csrf.enabled: true  # Set to 'false' to disable CSRF protection`
+
+**Environment variable:**  
+`CAMUNDA_SECURITY_CSRF_ENABLED=true  # Set to false to disable CSRF protection`
+
+:::caution
+Disabling CSRF protection is not recommended for production environments as it leaves your application vulnerable to cross-site request forgery attacks.
+:::
+
+## Authentication method differences
+
+### Basic authentication
+
+- CSRF tokens are generated upon successful login via the `/login` endpoint.
+- The token is included in the response header and stored in a secure cookie.
+- Subsequent requests must include the token in the `X-CSRF-TOKEN` header.
+- Tokens are cleared on logout.
+
+### OIDC (OpenID Connect) authentication
+
+- CSRF protection works similarly to Basic Auth.
+- Tokens are generated after successful OAuth2/OIDC authentication.
+- The same `X-CSRF-TOKEN` header mechanism applies.
+- Compatible with both JWT bearer tokens and session-based authentication.
+
+## Protected vs unprotected paths
+
+### Protected paths (require CSRF token)
+
+- `/api/**` – API endpoints (except specifically excluded paths)
+- `/v1/**`, `/v2/**` – Versioned API endpoints
+- All state-changing operations (POST, PUT, DELETE, PATCH)
+
+### Unprotected paths (no CSRF token required)
+
+- `/actuator/**` – Health and monitoring endpoints
+- `/swagger-ui/**` – API documentation
+- `/v2/license` – Public license endpoint
+- `/error` – Error handling
+- Authentication endpoints (`/login`, `/logout`)
+- Safe HTTP methods (GET, HEAD, OPTIONS, TRACE)
+
+## Implementation details
+
+### Request matching logic
+
+The implementation determines whether CSRF protection should be applied based on:
+
+- **HTTP method**: Safe methods (GET, HEAD, TRACE, OPTIONS) bypass CSRF checks.
+- **Request path**: Unprotected paths are excluded from CSRF validation.
+- **Session presence**: API calls with active browser sessions require CSRF tokens.
+- **Swagger UI**: Requests originating from Swagger UI are exempt.
+
+## Troubleshooting
+
+### Common issues
+
+**403 Forbidden on POST/PUT/DELETE requests**
```

Comment:
I suggest to add a hint here that this is only relevant when people have a session and that there can also be other, csrf-unrelated reasons for a 403. Otherwise, people will get stuck here with problems where the suggested solutions won't help.

Reply:
Added hint